src/indexApi.ts

```diff
@@ -2,17 +2,81 @@
 // =============================================================================
 
 import cluster from 'cluster'
+import { formatISO } from 'date-fns'
 import { forkChildren, setupDatabase } from 'edge-server-tools'
 
 import { config } from './utils/config'
-import { couchSchema } from './utils/couchSchema'
+import { couchSchema, minAmtDbPrefix } from './utils/couchSchema'
+import { ErrorResponse, makeErrorResponse } from './utils/errorResponse'
+import { cleanMinAmountDocs, findMinimum } from './utils/getMinimum'
+import { mockPlugins } from './utils/mockPlugins'
+import { checkQueryString, fetchSameKeyDocs } from './utils/utils'
 
 const express = require('express')
 const http = require('http')
+const nano = require('nano')
+const promisify = require('promisify-node')
+
+const RouteError: ErrorResponse = makeErrorResponse(
+  'bad_query',
+  404,
+  'Endpoint not found'
+)
 
 // call the packages we need
 const app = express()
 
+// Nano for CouchDB
+// =============================================================================
+const nanoDb = nano(config.dbFullpath)
+const minAmtDocScopeArr = mockPlugins.map(plugin => {
+  const pluginMinAmtDocScope = nanoDb.db.use(minAmtDbPrefix + plugin.pluginName)
+  return promisify(pluginMinAmtDocScope)
+})
+
+// ROUTES FOR OUR API
+// =============================================================================
+const router = express.Router()
+
+router.get('/getMinimum/', async function (req, res, next) {
+  try {
+    const currencyPairResult = checkQueryString(req.query.currencyPair) // Param of 'currencyPair' in URI
+    if (typeof currencyPairResult !== 'string') {
+      throw currencyPairResult // currencyPairResult is equal to ParamError
+    }
+    const currentUTCDate: string = formatISO(new Date(), {
+      representation: 'date'
+    }) // Variable for current UTC date using date-fns
+
+    const minAmtDocKey = currentUTCDate + ':' + currencyPairResult
+    const minAmtDocs = await fetchSameKeyDocs(minAmtDocKey, minAmtDocScopeArr)
+    const cleanMinAmtData = cleanMinAmountDocs(minAmtDocs)
+    const minAmtResult = findMinimum(cleanMinAmtData)
+
+    if (typeof minAmtResult !== 'string') {
+      throw minAmtResult // minAmtResult is equal to CurrencyPairDataError
+    }
```

Comment:
Place in a new function outside the middleware. The route will then be a one-liner inside the `try..catch`, which will reduce the number of lines for this middleware. NOTE: Try to keep functionality that uses `req, res, next` inside the middleware, and everything outside of it.